spec/tapioca/compilers/dsl/active_record_typed_store_spec.rb

```diff
@@ -0,0 +1,194 @@
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadatado do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+
+      RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for TypedStore classes with accessor") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: true)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T.nilable(String)) }
+          def reviewer; end
+
+          sig { params(reviewer: T.nilable(String)).returns(T.nilable(String)) }
+          def reviewer=(reviewer); end
+
+          sig { returns(T::Boolean) }
+          def reviewer?; end
+
+          sig { returns(T.nilable(String)) }
+          def reviewer_before_last_save; end
+
+          sig { returns(T.nilable([T.nilable(String), T.nilable(String)])) }
+          def reviewer_change; end
+
+          sig { returns(T::Boolean) }
+          def reviewer_changed?; end
+
+          sig { returns(T.nilable(String)) }
+          def reviewer_was; end
+
+          sig { returns(T.nilable([T.nilable(String), T.nilable(String)])) }
+          def saved_change_to_reviewer; end
+
+          sig { returns(T::Boolean) }
+          def saved_change_to_reviewer?; end
+        end
+       RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for TypedStore classes with fields null and default as false") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.boolean(:reviewed, null: false, default: false)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T::Boolean) }
+          def reviewed; end
+
+          sig { params(reviewed: T::Boolean).returns(T::Boolean) }
+          def reviewed=(reviewed); end
+
+          sig { returns(T::Boolean) }
+          def reviewed?; end
+
+          sig { returns(T::Boolean) }
+          def reviewed_before_last_save; end
+
+          sig { returns(T.nilable([T::Boolean, T::Boolean])) }
+          def reviewed_change; end
+
+          sig { returns(T::Boolean) }
+          def reviewed_changed?; end
+
+          sig { returns(T::Boolean) }
+          def reviewed_was; end
+
+          sig { returns(T.nilable([T::Boolean, T::Boolean])) }
+          def saved_change_to_reviewed; end
+
+          sig { returns(T::Boolean) }
+          def saved_change_to_reviewed?; end
+        end
+        RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for simple TypedStore classes with no fields specified ") do
```

Comment:
What is this testing that the `s.string(:reviewer, blank: false, accessor: true)` example isn't testing? They look the same to me.

Reply:
I will change the description to reflect test for `date: "Date"` type.